internal/controller/streams.go

```diff
@@ -570,8 +575,24 @@ func ConnectStreams(ctx context.Context, logger *zap.Logger, envMap EnvironmentC
 			ciliumFlowCollector := sm.findHubbleRelay(ctx, sm.streamClient.ciliumNamespace)
 			if ciliumFlowCollector == nil {
 				sm.streamClient.disableNetworkFlowsCilium = true
+				sm.streamClient.flowCollector = pb.FlowCollector_FLOW_COLLECTOR_FALCO
 			} else {
 				sm.streamClient.disableNetworkFlowsCilium = false
+				sm.streamClient.flowCollector = pb.FlowCollector_FLOW_COLLECTOR_CILIUM
+			}
+			// If neither is available
+			if sm.streamClient.flowCollector != pb.FlowCollector_FLOW_COLLECTOR_CILIUM &&
+				sm.streamClient.flowCollector != pb.FlowCollector_FLOW_COLLECTOR_FALCO {
+				sm.streamClient.flowCollector = pb.FlowCollector_FLOW_COLLECTOR_DISABLED
+			}
+			// Wait for the flow collector decision
+			sm.streamClient.flowCollectorWG.Done()
```

Comment:
I'm sorry, I gave you the wrong feedback.
Re-reading this, it turns out we don't need any synchronization, since we decide the flow collector before we even start the resources stream that will send the cluster metadata, cf. this line below:
https://github.com/illumio/cloud-operator/pull/165/files#diff-65d4066e2022c168a97a2083474d348fa6d5d1fffb6e8d7378282935ad187940R603
Therefore, we don't need to add this WaitGroup, we're already good without it.
```suggestion
```